CDCgov/prime-reportstream
# 9224 correct database entries when UP routing sends to no receivers

This PR makes changes that correct issues with the database entries written when the UP routing function sends to no receivers.

Test Steps:
1. Use the `Single HL7 Full ELR` test to submit data to be routed. Ensure the data sent does not have a valid receiver.
2. Retrieve the report `id` from the data returned from the API call.
3. Allow time for the data to be routed, then confirm the entry written for the `none` event on the `report_file` table contains an `item_count` of `0` and a `body_url` of `null`.

## Changes
- UP routing function no longer attempts to upload an Azure blob attached to the `none` event when there are no receivers.
- Update ActionHistory functions to allow creating reports without Azure blobs.
- Add additional tests for ActionHistory functions.
- Updated the test for UP routing no receivers.

## Checklist

### Testing

## Linked Issues
- #9224

## To Be Done
- `trackCreatedReport`'s overloads can probably be refactored into a single function. This function is used heavily, however, and we would want to consider how much of this function should remain when the legacy pipeline is retired. Ticket created: #9345
- The changes in this PR may inform the implementation of #8451
